From 6c1aef393ba4e3756f0f5d88e813a6b985c6efff Mon Sep 17 00:00:00 2001
From: Alexander Graf <agraf@suse.de>
Date: Sun, 12 Feb 2023 17:34:34 +0000
Subject: [PATCH] XXX openSUSE XXX: Load dtb from partition 2

Upstream U-Boot loads the device tree from the EFI partition. This is
mostly because upstream we don't know which partition really holds the
boot data.

For openSUSE however we do know, so let's just patch the boot script to
load device trees from partition 2.

Signed-off-by: Alexander Graf <agraf@suse.de>
---
 include/config_distro_bootcmd.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/include/config_distro_bootcmd.h b/include/config_distro_bootcmd.h
index c3a2414b..12486c86 100644
--- a/include/config_distro_bootcmd.h
+++ b/include/config_distro_bootcmd.h
@@ -155,17 +155,17 @@
 		"fi\0"                                                    \
 	\
 	"load_efi_dtb="                                                   \
-		"load ${devtype} ${devnum}:${distro_bootpart} "           \
+		"load ${devtype} ${devnum}:2 "           \
 			"${fdt_addr_r} ${prefix}${efi_fdtfile}\0"         \
 	\
-	"efi_dtb_prefixes=/ /dtb/ /dtb/current/\0"                        \
+	"efi_dtb_prefixes=/ /dtb/ /dtb/current/ /boot/ /boot/dtb/ /boot/dtb/current/\0" \
 	"scan_dev_for_efi="                                               \
 		"setenv efi_fdtfile ${fdtfile}; "                         \
 		BOOTENV_EFI_SET_FDTFILE_FALLBACK                          \
 		BOOTENV_RUN_EXTENSION_INIT                                \
 		"for prefix in ${efi_dtb_prefixes}; do "                  \
 			"if test -e ${devtype} "                          \
-					"${devnum}:${distro_bootpart} "   \
+					"${devnum}:2 "   \
 					"${prefix}${efi_fdtfile}; then "  \
 				"run load_efi_dtb; "                      \
 				BOOTENV_RUN_EXTENSION_APPLY               \
-- 
2.19.0

